src/irmin-test/store.ml

```diff
@@ -417,6 +421,76 @@ module Make (S : S) = struct
     in
     run x test
 
+  let test_closure x () =
```

Comment:
I find this test really hard to read. Perhaps you could split the graph initialisation step out into a helper function? Also the assertion steps might want to take strings explaining what is being asserted (rather than having comments nearby).

Reply:
I compacted the test, hopefully it is more clear.

Reply:
Yep, it reads much more clearly now; thanks!